src/scripts/main.js

```diff
@@ -1 +1,88 @@
 'use strict';
+
+let promiseNumber;
+let firstCheck = false;
+let secondCheck = false;
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    promiseNumber = 1;
+    resolve();
+  });
+
+  setTimeout(() => {
+    reject();
+  }, 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    promiseNumber = 2;
+    resolve();
+  });
+
+  document.addEventListener('contextmenu', (eve) => {
+    eve.preventDefault();
+    promiseNumber = 2;
+    resolve();
+  });
+});
+
+const promise3 = new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    firstCheck = true;
+
+    if (firstCheck && secondCheck) {
+      promiseNumber = 3;
+      resolve();
+    }
+  });
+
+  document.addEventListener('contextmenu', () => {
+    secondCheck = true;
+
+    if (firstCheck && secondCheck) {
+      promiseNumber = 3;
+      resolve();
+    }
+  });
+});
```

Comment:
The `promise3` promise resolves only when both `firstCheck` and `secondCheck` are true. Ensure that this logic aligns with the intended behavior, as it requires both a click and a context menu event to resolve.